src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/CoronaTestServiceTests.swift

```diff
@@ -1677,7 +1677,8 @@ class CoronaTestServiceTests: CWATestCase {
 		}
 		waitForExpectations(timeout: .short)
 
-		XCTAssertEqual(mockNotificationCenter.notificationRequests.count, 2)
+		// has to be 6 because each call to "updateTestResults()" add one notification for antigen and one for pcr
```

Comment:
Are you sure that it should be 6?
Another comment above states this "// Updating two more times to check that notification are only scheduled once".
Looks like the notifications are expected to only trigger once .

Reply:
Yes, it's because of the implementation of the `MockUserNotificationCenter`. It saves each notification request to the array when presenting it. Thats all the logic inside of it. So therefore it has to be 6 because `updateTestResults(presentNotification: true)` is called three times and that presents one notification for PCR and one for Antigen. The assumption it would only save 2 requests was wrong.

Reply:
This test is not testing MockUserNotificationCenter. The mock is used to figure out wether the CoronaTestService will trigger the right amount of notifications.

Looked into details of CoronaTestService and need to block this PR.
Your change is altering the expected behavior.
It could lead to redundant triggered notifications (and maybe also other unexpected behavior).

Reply:
This failing test also indicates, that the operations are not executed serially.